database/guests.ts

```diff
@@ -1,44 +1,70 @@
-declare module globalThis {
-  let guestsDatabase: Guest[];
-}
+import type { Guest } from '../migrations/00000-createTableGuests';
+import { sql } from './connect';
 
-// Initialize database to empty array only once
-//
-// TODO: Add a full PostgreSQL database connection
-if (!('guestsDatabase' in globalThis)) {
-  globalThis.guestsDatabase = [];
-}
-
-const guests = globalThis.guestsDatabase;
-
-console.log('guests in db', guests);
-
-export type Guest = {
-  id: number;
-  firstName: string;
-  lastName: string;
-  attending: boolean;
-};
-
-export function getGuests() {
+export const getGuestsInsecure = async () => {
+  const guests = await sql<Guest[]>`
+    SELECT
+      *
+    FROM
+      guests
+  `;
   return guests;
-}
+};
 
-export function getGuest(id: number) {
-  return guests.find((guest: Guest) => Number(guest.id) === Number(id));
-}
+export const getGuestInsecure = async (guestId: Guest['id']) => {
+  const [guest] = await sql<Guest[]>`
+    SELECT
+      *
+    FROM
+      guests
+    WHERE
+      id = ${guestId}
+  `;
+  return guest;
+};
 
-export function addGuest(guest: Guest) {
-  guests.push(guest);
-  return guests;
-}
+export const createGuestInsecure = async (newGuest: Omit<Guest, 'id'>) => {
+  const [guest] = await sql<Guest[]>`
+    INSERT INTO
+      guests (
+        first_name,
+        last_name,
+        attending
+      )
+    VALUES
+      (
+        ${newGuest.firstName},
+        ${newGuest.lastName},
+        ${newGuest.attending}
+      )
+    RETURNING
+      guests.*
+  `;
+  return guest;
+};
 
-export function deleteGuest(id: number) {
-  const index = guests.findIndex((guest) => guest.id === id);
+export const updateGuestInsecure = async (updatedGuest: Guest) => {
+  const [guest] = await sql<Guest[]>`
+    UPDATE guests
+    SET
+      first_name = ${updatedGuest.firstName},
+      last_name = ${updatedGuest.lastName},
+      attending = ${updatedGuest.attending}
+    WHERE
+      id = ${updatedGuest.id}
+    RETURNING
+      guests.*
+  `;
+  return guest;
+};
 
-  if (index === -1) {
-    return;
-  }
-  guests.splice(index, 1);
-  return guests;
-}
+export const deleteGuestInsecure = async (id: number) => {
+  const [guest] = await sql<Guest[]>`
+    DELETE FROM guests
+    WHERE
+      id = ${id}
```

Comment:
```suggestion
export const deleteGuestInsecure = async (guestId: Guest['id']) => {
  const [guest] = await sql<Guest[]>`
    DELETE FROM guests
    WHERE
      id = ${guestId}
```

Reply:
added in this commit 
[02412d3](https://github.com/upleveled/expo-example-spring-2024-atvie/pull/4/commits/02412d3489fef1d658e479ba88bfe6f07a8187a7)